### PR TITLE
Version 2 updates

### PR DESCRIPTION
**Pull request by TieDyeGeek**
* More Entity options
* CreateSetup() and UpdateSetup() functions to modify entities prior to creating / saving

### CODE REVIEW

**Review by jared-craig-cse — APPROVED** [review]

File: src/CSESoftware.Core/Entity/BaseEntity.cs [hunk 1]
```diff
@@ -1,22 +1,25 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Globalization;
 
 namespace CSESoftware.Core.Entity
 {
-    public abstract class BaseEntity<T> : IBaseEntity<T>
+    public abstract class BaseEntity<T> : EntityWithId<T>, IBaseEntity<T>
```
Comment (NickBehrends) [inline src/CSESoftware.Core/Entity/BaseEntity.cs:5]: Maybe I'm wrong, but it seems like only IModifiedEntity declares CreatedDate and ModifiedDate but we're using the properties here and we're not implementing this interface, am I missing something?

Comment (NickBehrends) [reply src/CSESoftware.Core/Entity/BaseEntity.cs:5]: And if that is correct, should we be moving the CreateSetup and UpdateSetup (is the "setup" suffix needed? Create and Update may flow easy enough) methods onto at least the IActiveEntity? seems like the lower interfaces are exposing it but the concrete classes aren't using it

Comment (jared-craig-cse) [reply src/CSESoftware.Core/Entity/BaseEntity.cs:5]: IBaseEntity has IModifiedEntity

Comment (NickBehrends) [reply src/CSESoftware.Core/Entity/BaseEntity.cs:5]: Ah I missed that, good catch

**Review by NickBehrends — APPROVED** [review]

**Review by brickgw — APPROVED** [review]

**Pull request merged**